src/wrapper_array.jl

```diff
@@ -67,9 +67,22 @@ Base.parent(x::NamedDimsArray) = x.data
 
 Returns a tuple of containing the names of all the dimensions of the array `A`.
 """
-names(::Type{<:NamedDimsArray{L}}) where L = L
-names(::Type{<:AbstractArray{T, N}}) where {T,N} = ntuple(_->:_, N)
-names(x::T) where T<:AbstractArray = names(T)
+names(x::NamedDimsArray{L}}) where L = L
+function names(x::T) where T<:AbstractArray
+    inner = parent(x)
+    # things that are not wrapper arrays are there own parents,
+    # so at that point we can give up looking for names
+    # also we have no way to deal with inner arrays of different dimensions, even if they do have names
+    # so give up then too
```

Comment:
maybe helpful to add an example (cos it's a bit of a weird case)? e.g.
> parent arrays can have a different number of dimensions,  e.g. the parent of a `Diagonal` is a `Vector`
we have no way to deal with the names in this case so give up then too